### ISSUE TITLE
`Array.__init__` is broken

### ISSUE THREAD

**Issue by sezelt** [py4dstem/py4DSTEM#369]
This line tries to set `self.shape` on an `Array` object, but shape is a property and refers to `self.data.shape` which cannot be changed.
https://github.com/py4dstem/py4DSTEM/blob/0631f28f833b0636c19828f59a3ed8207d4246dc/py4DSTEM/io/datastructure/emd/array.py#L198

Thus you get this error when you try to `__init__` an array with `slicelabels`:
```
File ~/git/py4DSTEM/py4DSTEM/io/datastructure/py4dstem/realslice.py:38, in RealSlice.__init__(self, data, name, pixel_size, pixel_units, slicelabels)
     35 if type(pixel_units) is not list: pixel_units = [pixel_units,pixel_units]
     37 # initialize as an Array
---> 38 Array.__init__(
     39     self,
     40     data = data,
     41     name = name,
     42     units = 'intensity',
     43     dims = [
     44         pixel_size[0],
     45         pixel_size[1],
     46     ],
     47     dim_units = [
     48         pixel_units[0],
     49         pixel_units[1],
     50     ],
     51     dim_names = [
     52         'Rx',
     53         'Ry'
     54     ],
     55     slicelabels = slicelabels
     56 )
     58 # setup the size/units with getter/setters
     59 self._pixel_size = pixel_size

File ~/git/py4DSTEM/py4DSTEM/io/datastructure/emd/array.py:198, in Array.__init__(self, data, name, units, dims, dim_names, dim_units, slicelabels)
    196 else:
    197     self.depth = self.shape[-1]
--> 198     self.shape = self.shape[:-1]
    199     dim_in_pixels = dim_in_pixels[:-1]
    200     self.rank -= 1

AttributeError: can't set attribute
```

I don't know what the intention was with having the shape not include the last dimension, so I don't want to touch it here and have some assumptions elsewhere be violated.

**Comment by bsavitzky** [comment]
Ah, interesting.....thanks Steve.  I will try to patch this tomorrow.  This logic is for Array stacks - i.e. for some N+1 dimensional array, the first N dimensions are treated normally, and the final dimension indexes distinct and optionally named N-D arrays which all share the same dimensions/units/calibrations.  In these cases we're calling N the `rank` of the N+1 dimensional Array, and the `.shape` property was intended to describe the dimensionality of those first N dimensions.  I think this should effect other code in only a few other fairly superficial places, but will have to look into it.

**Comment by bsavitzky** [comment]
Solved by PR #370
